packages/flutter/lib/src/material/input_border.dart

```diff
@@ -116,6 +116,92 @@ class _NoInputBorder extends InputBorder {
   }
 }
 
+/// Clips an [InputDecorator]'s [InputDecoration.filled] background to a rounded
+/// rectangle.
+///
+/// This class is similar to [OutlineBorder] except that no outline is drawn.
+///
+/// See also:
+///
+///  * [UnderlineInputBorder], the default [InputDecorator] border which
+///    draws a horizontal line at the bottom of the input decorator's container.
+///  * [OutlineInputBorder], an [InputDecorator] border which draws a
+///    rounded rectangle around the input decorator's container.
+///  * [InputDecoration], which is used to configure an [InputDecorator].
+class FilledInputBorder extends InputBorder {
+  /// Clips the [InputDecoration.filled] background of an [InputDecorator] to a
+  /// rounded rectangle.
+  ///
+  /// The [borderRadius] parameter defaults to a value where all corners have a
+  /// circular radius of 4.0. The [borderRadius] parameter must not be null.
+  const FilledInputBorder({
+    this.borderRadius = const BorderRadius.all(Radius.circular(4.0)),
+  }) : assert(borderRadius != null),
+        super(borderSide: BorderSide.none);
+
+  /// The radii of the border's rounded rectangle corners.
+  ///
+  /// When this border is used with a filled input decorator, see
+  /// [InputDecoration.filled], the border radius defines the shape
+  /// of the background fill.
+  ///
+  /// By default all corners have a circular radius of 4.0.
+  final BorderRadius borderRadius;
+
+  @override
+  bool get isOutline => false;
+
+  @override
+  InputBorder copyWith({BorderSide borderSide, BorderRadius borderRadius}) {
+    return FilledInputBorder(
+      borderRadius: borderRadius ?? this.borderRadius,
+    );
+  }
+
+  @override
+  EdgeInsetsGeometry get dimensions => EdgeInsets.zero;
+
+  @override
+  FilledInputBorder scale(double t) {
+    return FilledInputBorder(borderRadius: borderRadius);
+  }
+
+  @override
+  Path getInnerPath(Rect rect, { TextDirection textDirection }) {
+    return Path()
+      ..addRect(Rect.fromLTWH(rect.left, rect.top, rect.width, math.max(0.0, rect.height - borderSide.width)));
+  }
+
+  @override
+  Path getOuterPath(Rect rect, { TextDirection textDirection }) {
+    return Path()..addRRect(borderRadius.resolve(textDirection).toRRect(rect));
+  }
+
+  @override
+  void paint(Canvas canvas, Rect rect, {
+    double gapStart,
+    double gapExtent = 0.0,
+    double gapPercentage = 0.0,
+    TextDirection textDirection,
+  }) {
+    // Do not paint.
+  }
+
+  @override
+  bool operator ==(dynamic other) {
+    if (identical(this, other))
+      return true;
+    if (runtimeType != other.runtimeType)
+      return false;
+    final FilledInputBorder typedOther = other;
+    return typedOther.borderSide == borderSide
+        && typedOther.borderRadius == borderRadius;
+  }
+
+  @override
+  int get hashCode => borderSide.hashCode;
```

Comment:
borderRadius.hashCode

Please add a simple test that verifies that == and hashCode work correctly (you might as well check all of the InputBorder classes).

Reply:
Put in some. Let me know if that's comprehensive enough.

Reply:
I didn't see any tests for operator ==?

Reply:
Oops, there they are!